### ISSUE TITLE
[All] set proper species:name & reaction:name in SBML files

### ISSUE THREAD

**Issue by matthiaskoenig** [whole-cell-tutors/whole-cell-reduced#20]
I looked at the other submodules and most of you just set the name attributes to the id. These makes it very difficult to make sense of the species and reactions. The names should be human readable and understandable names.

For instance in translation

```
<species id="TU_278__c" name="TU_278__c" compartment="c" initialAmount="1" substanceUnits="item" hasOnlySubstanceUnits="true" boundaryCondition="false" constant="false"/>
      <species id="MG_266_MONOMER__c" name="MG_266_MONOMER__c" compartment="c" initialAmount="1" substanceUnits="item" hasOnlySubstanceUnits="true" boundaryCondition="false" constant="false"/>
      <species id="MG_070_MONOMER__c" name="MG_070_MONOMER__c" compartment="c" initialAmount="1" substanceUnits="item" hasOnlySubstanceUnits="true" boundaryCondition="false" constant="false"/>
```

You should write the names provided by Karr in the database, which makes the SBML much more understandable for the other groups. For instance I have no idea what TU_278__c is.

In Metabolism this looks than like

```
<species id="MG_351_HEXAMER" name="inorganic pyrophosphatase" compartment="c" initialAmount="0" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
      <species id="MG_357_DIMER" name="acetate kinase" compartment="c" initialAmount="0" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
      <species id="MG_368_DIMER" name="fatty acid/phospholipid synthesis protein PlsX" compartment="c" initialAmount="0" hasOnlySubstanceUnits="false" boundaryCondition="false" constant="false"/>
```

If anyone needs a mapping
id -> name
I can create one on demand from the knowledgebase.

**Comment by manikitos** [comment]
What should we do with empty names?

**Comment by matthiaskoenig** [comment]
Hi Nikita,
everything with a wid, i.e. Karr id, has a name. I added the full mapping of wid -> names in 
https://github.com/whole-cell-tutors/whole-cell-reduced/blob/master/Integration/entry_names.csv

If you create SBML objects not present in the Karr model you should create some human understandable name which explains the object.

**Comment by manikitos** [comment]
This means that tables from the GoogleDrive are not complete: https://drive.google.com/a/developmentontheedge.com/folderview?id=0B7b8ClM7CZMrOTJ1WE05QklyRjA&usp=sharing_eid&ts=56140f0e&tid=0B7b8ClM7CZMrMFA0ek83ZHNINkE
In your table also are rows with empty name column, e.g. MG_028_MONOMER. Or I don't understand something?
I didn't mean objects which are not present in the Karr model, they of course must have human understandable names.

**Comment by matthiaskoenig** [comment]
Yes, for MG_028_MONOMER there is nothing in the data base for the name attribute. So I have no idea what it is. I would recommend for this cases to just name it
"model_type wid"
i.e.
"ProteinMonomer MG_028_MONOMER"
This is all the information available about the entry in the database.

**Comment by binfalse** [comment]
see commits 80c7182a9da89cfabf17dce4d9788213584eaa3a and c2fc8c44042293e452e1757d0008363909bf22f0 which fix this for species in [ProteinDecay] and [Translation]

**Comment by manikitos** [comment]
Fixed in commit 638bb48 for species in [ProteinTranslocation]

**Comment by manikitos** [comment]
Fixed in commit d869688 for reactions in [ProteinTranslocation]

**Comment by manikitos** [comment]
As I can see it's fixed.
